### DISCUSSION TITLE
How to change the lyrics panel?

### DISCUSSION THREAD

**Discussion by Stealth3** [TT-ReBORN/Georgia-ReBORN#44] [Q&A] [answered]
I can't figure out how to bring up the options for scrolling / font size/ alignment etc for the lyrics panel. Any idea how to change these things?

**Comment by silverwings15** [comment]
you can adjust font size here 
![image](https://user-images.githubusercontent.com/10742312/148568797-db990956-754f-4dc9-857f-de08029d176f.png)

not sure about other two. while on the topic of lyrics panel, i also have to note that the auto scrolling is quite choppy, but that's probably a limitation of the addon which i understand is very old and has been abandoned for sometime. wonder if foo_openlyrics (https://github.com/jacquesh/foo_openlyrics) is an all around improvement, and if it is maybe we can see it implemented in Georgia/G-R in the future

**Comment by TT-ReBORN** [comment, marked as answer]
Hi @Stealth3,

there are no options for scrolling and alignment. Synced lyrics and unsynced lyrics will scroll automatically and accordingly to track progress.
You can only change the font size as @silverwings15 already shown and other things in File > Preferences > Tools > Lyric Show 3 
( for unexperienced users, I would not recommended changing things here ). 

Also if you display synced lyrics which are highlighted with yellow/neon marker and the timestamps are not correct, you can change the offset via mouse wheel scroll on album art to adjust it. If timestamps are correct, offset should be always 0.

@silverwings15 I don't know what you mean auto scrolling is quite choppy? Is your offset 0 for synced lyrics?
For me, auto scrolling is super smooth, tested it on an old computer, HD though. Maybe in 4K on old slow CPU's?
Of course I know foo_openlyrics and I follow its progress, would love to replace it but not quite yet.
It will be replaced in the future, once it can also replace foo_multisource which has great lyric server sources 
and most lyrics are found because of it. Also foo_openlyrics does not currently support radio streams...

-TT

**Reply by Stealth3** [reply]
I can click on lyrics panel 3 and change the settings I want, but they do not change in the lyrics panel shown. Normally when using lyrics panel 3 I can right click on the panel and change the things I want, but your skin has its own right click menu I think.

**Reply by TT-ReBORN** [reply]
Yes that's right, the Lyrics Panel 3 component is only used for fetching lyrics with foo_multisource, it is basically hidden.
See for yourself in File > Preferences > Display > Columns UI > Layout > Vertical splitter ( Hidden checked ).

But you can find the same Lyric Panel 3 settings if you shift + click on top menu 'View' > Lyric Show 3. Visual appearance settings will be ignored though...

**Reply by Stealth3** [reply]
Thanks. I have another problem. I added a folder and it shows up in library view as artist and then you click on the artist and you see the album. Looks good, but now when I go to add music it doesn't show up in the library view, it doesn't show anywhere. How do I add more music to my library view. I just want to be able to add folders and have them show up like the first time when it worked.

**Reply by TT-ReBORN** [reply]
Do you mean the Library? Because "Library View" is the auto-created playlist when you double click an album/track from the library.

If you mean the Library, then go to File > Preferences > Media Library . Add your folders or just right click on the path and hit Rescan now.

If you mean the auto-created playlist named "Library View" this will be always overwritten by new albums/tracks once you double click from Library. You can right click in Library and click "Add to current playlist" or "Send to new playlist". But the best way to create your own playlist would be to manually create new playlists, in playlist right click > Create new playlist. Now you can drag and drop music files into it. It will not be overwritten from the library. You can change and manage playlists through the playlist manager. Right click in playlist > Playlist manager or you can click on the playlist top on the current playlist name ( mouse hover at the top of the playlist ). And if you switch to your own created playlist and go library, you can use the right click "Add to current playlist" to also fill the playlist. Be sure your playlists are active and not the "Library View" playlist.

**Reply by Stealth3** [reply]
When I go to File > Add Folder / Add File and pick something it doesn't show up anywhere in the player. Is what I mean. Where does that music go that i add so I can listen to it? If I click on the Library button at the top [Details / Library / Biography / Lyrics] it shows the first album I ever added from File > add folder but doesnt show anything else i add.

Edit: i realize now library is showing music from the default music folder, but still dont know where music goes that i add from File > Add

**Reply by TT-ReBORN** [reply]
If you do it the old fashion way ( File > Add Folder / Add File ) it will automatically go to your active playlist. You can also drag and drop your music folders to the playlist. The Playlist is the HOME, that means no active panel in the top menu should be active. 
For example, if you select Library in the top menu, click again on the Library button to deactivate it and it will automatically show the playlist. Biography and Lyrics panel also shows the playlist.

Edit: Also if the Library is active you can not drag and drop music folders there and the command File > Add Folder / Add File
will not be added to the Library only to the Playlist. Also be sure to have NOT created an auto playlist but just a normal regular playlist. Auto-playlists are locked and used for the Library music collection.

**Reply by Stealth3** [reply]
Yeah I must have did something because no music that i added shows when deactivated a pressed button like library or lyrics.

**Reply by TT-ReBORN** [reply]
Can't think of anything other than you created a auto-playlist. See if there is a lock symbol if you mouse hover the playlist manager text button at the top of the playlist. If there is, right click on the playlist and "Create a new playlist" or just hit CTRL+N.

**Reply by Stealth3** [reply]
Thanks you solved it!

**Comment by silverwings15** [comment]
@TT-ReBORN i guess it's not choppy per se, but it's not 60 FPS smooth either. i screenrecorded it next to spotify's lyric panel so you can see what i mean

https://gfycat.com/disloyaloccasionalindianjackal

**Reply by TT-ReBORN** [reply]
Thanks for the video, what player size are you using? Also do you use 4k? 
Because for me in HD res and player size 'Small' to 'Normal' it's much smoother. I guess it's just the GDI+ rendering, it has bad performance.
@TheQwertiest wanted to replace it, it is on his TODO list, see here: https://github.com/TheQwertiest/foo_spider_monkey_panel/projects/1#card-28650466

**Reply by silverwings15** [reply]
i'm using ReBORN at full screen on a 2560x1440p panel @ 150% scaling, 14 inch laptop, 4k option is chosen as autodetected. funnily enough manually changing the Player Size settings or enabling/disabling 4k mode seem to have no effect on my setup. good to know that improvements are forthcoming!

edit: okay nevermind, enabling/disabling 4k does have an effect, but only in that it changes my font settings? i'm manually setting values for each component anyway to get optimal scaling on my setup.

**Reply by TT-ReBORN** [reply]
Hey what a coincidence really! 

QHD res was on my TODO list because I've encountered more and more users who are using QHD monitors. Actually I have yesterday finished to make an extra QHD option for Georgia-ReBORN which will have new optimized sizes ( 9 sizes, Player Size 'Small', 'Normal' and 'Large' for Default, Artwork, Compact mode ). This was quite a task because I needed to refactor my whole auto-detect and resolution code... I'm quite happy with the outcome, code is much cleaner and better and QHD users will finally enjoy. I've have just today send my new version another beta tester who is testing it with his QHD monitor.

**Reply by silverwings15** [reply]
haha, now i can't go back to my font settings like before i started enabling/disabling 4k. would be happy to help you test out your new version then if you need

**Reply by TT-ReBORN** [reply]
Do you have an account on HydrogenAudio? 
I don't want to populate this unrelated conversion here. Best way is to send me a PM on HydrogenAudio 
and I can send you the QHD version for testing...

My account on HydrogenAudio is this:
https://hydrogenaud.io/index.php?action=profile;u=139848

Edit: Also font settings are no problem, I have optimized it for QHD users.
